cassovary-core/src/main/scala/com/twitter/cassovary/graph/Node.scala

```diff
@@ -273,9 +289,15 @@ trait Node {
 
 object Node {
   private lazy val randGen: Random = new Random
-  def apply(nodeId: Int, out: Seq[Int], in: Seq[Int]) = new Node {
-    val id = nodeId
-    def inboundNodes() = in
-    def outboundNodes() = out
+
+  def apply(nodeId: Int, out: Seq[Int], in: Seq[Int]): Node = {
+    new SeqBasedNode(nodeId, in, out)
+  }
+
+  def withSortedNeighbors(nodeId: Int, in: Array[Int], out: Array[Int]) = {
+    new SeqBasedNode(nodeId, in, out) with SortedNeighborsNodeOps
   }
 }
+
+class SeqBasedNode private[graph] (val id: Int, val inboundNodes: Seq[Int], val outboundNodes: Seq[Int])
```

Comment:
It's only a constructor, not any logic at all.

I have it only in order not to write:
```
new Node {
  val id = nodeId
  override def inboundNodes() = in
  overrided def outboundNodes() = out
}
```